src/qml/RelationCombobox.qml

```diff
@@ -334,102 +344,210 @@ Item {
+            height: fontMetrics.height + 12
+            text: useCompleter ? completer : comboBox.displayText
+            font: comboBox.font
+            horizontalAlignment: Text.AlignLeft
+            verticalAlignment: Text.AlignVCenter
+            textFormat: Text.RichText
+            clip: true
+            elide: Text.ElideRight
+
+            color: value === undefined || !enabled ? 'gray' : searchableText.text === '' ? 'black' : 'gray'
+        }
 
-      Text {
-        padding: 5
-        width: parent.width - dropDownArrowCanvas.width - dropDownArrowCanvas.anchors.rightMargin * 2
-        text: comboBox.displayText
-        font: comboBox.font
-        horizontalAlignment: Text.AlignLeft
-        verticalAlignment: Text.AlignVCenter
-        elide: Text.ElideRight
-        color: value === undefined || !enabled ? 'gray' : 'black'
-      }
+        TextField  {
+            id: searchableText
+
+            property string typedFilter: ''
+
+            anchors.verticalCenter: parent.verticalCenter
+            padding: 5
+            topPadding: fontMetrics.ascent - 1
+            topInset: 0
+            bottomInset: 0
+            width: parent.width - dropDownArrowCanvas.width - dropDownArrowCanvas.anchors.rightMargin * 2
+            text: ''
+            font: comboBox.font
+            horizontalAlignment: TextInput.AlignLeft
+            verticalAlignment: TextInput.AlignVCenter
+
+            color: 'black'
+            background: Rectangle {
+                color: "transparent"
+                border.color: "transparent"
+                border.width: 0
+            }
 
-      Canvas {
-        id: dropDownArrowCanvas
-        anchors.right: parent.right
-        anchors.rightMargin: 10
-        anchors.verticalCenter: parent.verticalCenter
-        width: 10
-        height: 5
-        contextType: "2d"
-
-        onPaint: {
-          if (!context) {
-            return
-          }
-            context.reset();
-            context.moveTo(0, 0);
-            context.lineTo(width, 0);
-            context.lineTo(width / 2, height);
-            context.closePath();
-            context.fillStyle = !enabled ? 'gray' : 'black'
-            context.fill();
+            inputMethodHints: Qt.ImhNoPredictiveText | Qt.ImhNoAutoUppercase | Qt.ImhSensitiveData
+
+            onDisplayTextChanged: {
+                if (activeFocus) {
+                    if (text != comboBox.displayText) {
+                        var trimmedText = text.trim();
+                        var matches = featureListModel.findDisplayValueMatches(trimmedText)
+                        if (matches.length > 0) {
+                            searchableLabel.completer = '<span style="color:rgba(0,0,0,0);">' + text + '</span><span style="font-weight:' + (matches.length === 1 ? 'bold' : 'normal' ) + ';">'  + featureListModel.dataFromRowIndex(matches[0], featureListModel.DisplayStringRole).substring(trimmedText.length) + '</span>'
```

Comment:
```suggestion
                            var remainder = featureListModel.dataFromRowIndex(matches[0], featureListModel.DisplayStringRole)
                              .substring(trimmedText.length)
                            searchableLabel.completer = ''
                                + '<span style="color:rgba(0,0,0,0);">' + text + '</span>'
                                + '<span style="font-weight:' + (matches.length === 1 ? 'bold' : 'normal' ) + ';">'  + remainder + '</span>'
```

Reply:
@suricactus , good one, that line was a bit intense :wink: